frontend/packages/console-app/src/components/cloud-shell/CloudShell.tsx

```diff
@@ -1,21 +1,51 @@
 import * as React from 'react';
-import { Button } from '@patternfly/react-core';
-import CloudShellBody from './CloudShellBody';
+import { Dispatch } from 'redux';
+import { connect } from 'react-redux';
+import { confirmModal } from '@console/internal/components/modals/confirm-modal';
+import { RootState } from '@console/internal/redux';
+import { isCloudShellExpanded } from '../../redux/reducers/cloud-shell-reducer';
+import { toggleCloudShellExpanded } from '../../redux/actions/cloud-shell-actions';
 import CloudShellDrawer from './CloudShellDrawer';
+import CloudShellTerminal from './CloudshellTerminal';
 
-const CloudShell: React.FC = () => {
-  const [open, setOpen] = React.useState(false);
+type StateProps = {
+  open?: boolean;
+};
+
+type DispatchProps = {
+  onClose: () => void;
+};
+
+type CloudShellProps = StateProps & DispatchProps;
+
+const CloudShell: React.FC<CloudShellProps> = ({ open, onClose }) => {
   return (
-    <>
-      {/* Remove this button once actual terminal is in place */}
-      <Button variant="control" onClick={() => setOpen(!open)}>
-        Open Drawer
-      </Button>
-      <CloudShellDrawer open={open} onClose={() => setOpen(false)}>
-        <CloudShellBody />
-      </CloudShellDrawer>
-    </>
+    <CloudShellDrawer open={open} onClose={onClose}>
+      <CloudShellTerminal />
+    </CloudShellDrawer>
   );
 };
 
-export default CloudShell;
+const stateToProps = (state: RootState): StateProps => ({
+  open: isCloudShellExpanded(state),
+});
+
+const dispatchToProps = (dispatch: Dispatch): DispatchProps => ({
+  onClose: () =>
+    confirmModal({
```

Comment:
This should not be part of redux callbacks, create a sperate callback for modal something like this 
```
const closeCloudShellConfirmationModal = (callback) => {confirmModal({    
      title: 'Close Terminal?',
      message: (
        <>
          This will close the terminal session. Content in the terminal will not be resolved in the
          next seesion.
        </>
      ),
      btnText: 'Yes',
      submitDanger: true,

      cancelText: 'No',
      executeFn: callback,
    })};
```
you can use the same in CloudShellMastheadButton as well.

Reply:
updated this